pkg/upgrade/machine_creator.go

```diff
@@ -77,9 +79,23 @@ func NewMachineCreator(options ...MachineCreatorOption) *MachineCreator {
 	return creator
 }
 
+func (n *MachineCreator) createMachine(machine *clusterapiv1alpha2.Machine) (*clusterapiv1alpha2.Machine, error) {
```

Comment:
there two methods can go away since we can fake the controller runtime client easily enough now